float64 hour
float64 minute